### PR TITLE
Rework monitoring-satellite on previews

### LINKED ISSUE #7544: Support `with-Observability`
Preview environments on Harvester should support "observability" (aka have monitoring-satellite installed).

Proposed Solution:
a) always install monitoring-satellite
b) install monitoring-satellite only when the flag `with-Observability` is passed to the Werft job.

### PR DESCRIPTION
**Pull request by ArthurSens**
## Description

This PR makes quite a few changes:

1. Install Monitoring-satellite on Harvester preview environments. Please notice that they are installed asynchronously with Gitpod and there is no opt-in. They are always deployed together.  Commit cec05edbd86973ae475a1e71d57cda09afb9f5fb
2. Make Prometheus and Grafana available through the VMs node port, making them available through Harvester's proxy. Commit b5c2a1208cda2a417e192a87544da373f9f24ec1
3. From a discussion with @mads-hartmann, we're also removing the stable DNS from monitoring satellite on previews. Doesn't matter if it's core-dev or Harvester. (Related: https://github.com/gitpod-io/observability/pull/76). Commit: a9639538541bc5e0f53337fc1519a358ff00259c
4. Provide utility scripts that can make Prometheus and Grafana's UI available without effort for both core-dev and harvester previews, replacing the previous approach of a stable DNS (requires engineers to access a workspace though). Commit bc8f663b40aedb85e9f8fe2323b39964140e79f8

We want to get stable DNS in the future, but they've been post-poned(See https://github.com/gitpod-io/ops/issues/881)

## Related Issue(s)

Fixes #7544 
Fixes https://github.com/gitpod-io/observability/issues/56

## Release Notes

```release-note
NONE
```

## Documentation

Yes it does, I'm still updating! I'll remove the draft mode once docs are up to date

### CODE REVIEW

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please ask for approval from **arthursens** and additionally assign **corneliusludmann** after the PR has been reviewed.
You can assign the PR to them by writing `/assign @corneliusludmann` in a comment when ready.

Associated issue: *#881*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/gitpod-io/gitpod/blob/main/OWNERS)**
- **[.werft/observability/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/.werft/observability/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by ArthurSens** [comment]
I'm just gonna grab lunch real quick and then update documentation :)

should be good to merge after that

File: .werft/build.ts
Comment (mads-hartmann) [inline .werft/build.ts:611]: I think needs to be guarded by withVM right? Otherwise we'd try to install monitoring in core-dev when you use the installer which means we'd almost always install monitoring as the installer is the default.

We could also move this some line [316 - 351](https://github.com/gitpod-io/gitpod/blob/main/.werft/build.ts#L316-L351) so that installing monitoring is part of "creating the VM" as disconnected from the Gitpod installation ☺️

**Review by mads-hartmann — COMMENTED** [review]
This is exciting! I left one comment for now but didn't have a chance to try it out yet. Will try this evening

**Comment by mads-hartmann** [comment]
@ArthurSens when I tried using with-helm and with-observability my node exporter had problems. I didn't debug it yet though so could be totally unrelated to these changes :) See job here ([first failure](https://werft.gitpod-dev.com/job/gitpod-build-mads-vm-monitoring.1/logs), [second failure](https://werft.gitpod-dev.com/job/gitpod-build-mads-vm-monitoring.2/logs)) I started from a branch that mirrors yours

**Comment by ArthurSens** [comment]
> @ArthurSens when I tried using with-helm and with-observability my node exporter had problems. I didn't debug it yet though so could be totally unrelated to these changes :) See job here ([first failure](https://werft.gitpod-dev.com/job/gitpod-build-mads-vm-monitoring.1/logs), [second failure](https://werft.gitpod-dev.com/job/gitpod-build-mads-vm-monitoring.2/logs)) I started from a branch that mirrors yours

Hmmm, that's bad but I do believe it's just because of instabilities of core-dev :/

Node exporter is a daemonset and thus runs on all nodes... adding more nodes doesn't help to make things stable for node-exporter. Opening a workspace today and looking at the pods running, node-exporter rollout has finally finished

**Review by mads-hartmann — APPROVED** [review]
@ArthurSens I tried the with VM option and your script. Worked like a charm 🚀

**Comment by roboquat** [comment]
LGTM label has been added.  <details>Git tree hash: d8a718010fb0fd362458a540b850df172cab4815</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/7871#pullrequestreview-865906956" title="Approved">mads-hartmann</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **corneliusludmann** after the PR has been reviewed.
You can assign the PR to them by writing `/assign @corneliusludmann` in a comment when ready.

Associated issue: *#881*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/gitpod-io/gitpod/blob/main/OWNERS)**
- ~~[.werft/observability/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/.werft/observability/OWNERS)~~ [mads-hartmann]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

Approval requirements bypassed by manually added approval.

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/7871#pullrequestreview-865906956" title="Approved">mads-hartmann</a>*

Associated issue: *#881*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/gitpod-io/gitpod/blob/main/OWNERS)**
- ~~[.werft/observability/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/.werft/observability/OWNERS)~~ [mads-hartmann]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
